uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/components/leave-type-wizard.component.js

```diff
@@ -270,6 +334,50 @@ define([
       });
     }
 
+    /**
+     * Sets values to the field map from the loaded Absence Type.
+     * - At first, it simply overrides values with loaded ones, however:
+     *   - if the default value is boolean, it transforms "1" to true, else to false
+     *   - if the loaded value is a whole decimal, it transform to a whole number
+     *   - if the loaded value is a decimal, it only leaves 1 decimal digit
+     * - The Category field is then set as a name from the value
+     * - If the Default Entitlement field is "0.00", it set to an empty string
+     */
+    function initExistingValues () {
+      _.each(vm.fieldsIndexed, function (field) {
+        var loadedValue = absenceType[field.name];
+
+        if (loadedValue === undefined) {
+          return;
+        }
+
+        if (typeof (field.defaultValue) === 'boolean') {
+          field.value = loadedValue === '1';
+        } else if (/^\d+\.[0]+$/.test(loadedValue)) {
+          field.value = loadedValue.replace(/^(\d+)\..+$/, '$1');
+        } else if (/^\d+\.\d{2,}$/.test(loadedValue)) {
+          field.value = loadedValue.replace(/^(\d+\.\d).+$/, '$1');
+        } else {
+          field.value = loadedValue;
+        }
+      });
+
+      vm.fieldsIndexed.category.value =
+        _.find(vm.leaveTypeCategories, { value: absenceType.category }).name;
+
+      if (vm.fieldsIndexed.default_entitlement.value === '0') {
+        vm.fieldsIndexed.default_entitlement.value = '';
+      }
+
+      if (vm.fieldsIndexed.accrual_expiration_duration.value) {
+        vm.fieldsIndexed.accrual_never_expire.value = false;
+      }
+
+      if (vm.fieldsIndexed.carry_forward_expiration_duration.value) {
+        vm.fieldsIndexed.carry_forward_expiration_duration_switch.value = true;
+      }
+    }
```

Comment:
I think this is more code, but explains what's going on with the primitive value transformation. And, if for some reason we need to consider another type of field, it can just be added to the array.

```js
function initExistingValues () {
  parsePrimitiveValues();
  runFieldValuesTransformations(); // the other ones
}

function parsePrimitiveValues () {
  var primitives = [ BooleanPrimitive, NumberPrimitive, DecimalPrimitive ];

  _.each(vm.fieldsIndexed, function (field) {
    var originalValue = absenceType[field.name];

    if (originalValue === undefined) {
      return;
    }

    field.value = originalValue;

    primitives.some(function (primitive) {
        if (primitive.test(field, originalValue)) {
          field.value = primitive.parse(originalValue);

          return true;
        }
    });
  });
}

var BooleanPrimitive = {
  test: function (field, value) {
    return typeof (field.defaultValue) === 'boolean';
  },
  parse: function (value) {
    return value === '1';
  }
};
```

Another alternative is to add the type to the fields list and just use the parse:

```js
// ...
fields: [
  {
    name: 'add_public_holiday_to_entitlement',
    defaultValue: false,
    hidden: true,
    type: 'Boolean'
  },
  // ...
]
// ...

var valueParses = {
  Boolean: function (value) {
    return value === '1';
  }
};

function parseValues () {
  _.each(vm.fieldsIndexed, function (field) {
    var originalValue = absenceType[field.name];
    var valueParser = valueParses[field.type];

    if (originalValue === undefined) {
      return;
    }

    field.value = originalValue;

    if (valueParser) {
      field.value = valueParser(field.value);
    }
  });
}
```

Reply:
Agreed, I will try to refine it.

Reply:
Done. I went with a second approach.